src/CMakeLists.txt

```diff
@@ -54,7 +54,7 @@ else(no_yasm)
     set(CMAKE_REQUIRED_QUIET ${save_quiet})
     if(not_arch_x32)
       message(STATUS " we are not x32")
-      set(HAVE_GOOD_YASM_ELF64 1)
+      set(HAVE_GOOD_YASM_ELF64 1 PARENT_SCOPE)
```

Comment:
we have following code in `src/CMakeLists.txt`

```cmake
if(HAVE_GOOD_YASM_ELF64)
  set(yasm_srcs
    common/crc32c_intel_fast_asm.S
    common/crc32c_intel_fast_zero_asm.S)
endif(HAVE_GOOD_YASM_ELF64)
```

which checks `HAVE_GOOD_YASM_ELF64`. but if we **only** `set(HAVE_GOOD_YASM_ELF64 1 PARENT_SCOPE)`. the local `HAVE_GOOD_YASM_ELF64` is not changed. so  `yasm_srcs` will be not defined and empty. that's why i'd suggest move the variable of `HAVE_GOOD_YASM_ELF64 ` down to `src/CMakeLists.txt` without worrying about its lexical scope.